helm-service/controller/onboarder.go

```diff
@@ -248,14 +99,18 @@ func (o *onboarder) OnboardGeneratedChart(helmManifest string, event keptnv2.Eve
 		if err := o.namespaceManager.InjectIstio(event.Project, event.Stage); err != nil {
 			return nil, err
 		}
-	} else {
+	} else if strategy == keptnevents.Direct {
 		o.getKeptnHandler().Logger.Debug(fmt.Sprintf("For service %s in stage %s with deployment strategy %s, a mesh chart is generated",
 			event.Service, event.Stage, strategy.String()))
 		generatedChart, err = o.chartGenerator.GenerateMeshChart(helmManifest, event.Project, event.Stage, event.Service)
 		if err != nil {
 			o.getKeptnHandler().Logger.Error("Error when generating the managed chart: " + err.Error())
 			return nil, err
 		}
+	} else {
```

Comment:
`else` is always very implicit. Please add a comment for which strategies the following code is executed.

Reply:
Actually, you cannot know. Its everything except bluegreen and direct ;) I would not mention this in a comment.

Reply:
True, could be anything, but it's implemented with `user-managed` in mind. How about
```
// No need to generate further charts for other strategies, e.g., user-managed
```